compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEPropagateDataLayout.cpp

```diff
@@ -15,6 +16,52 @@ namespace mlir::iree_compiler::AMDAIE {
 
 namespace {
 
+/// Forces `outs` operands of linalg operations to use `tensor.empty` if the
+/// value of the `outs` operand is not used within the op.  This is only
+/// implemented for `linalg.generic` operations for now, but should hold for all
+/// linalg structured ops.
+struct RemoveOutsDependency : public OpRewritePattern<linalg::GenericOp> {
+  using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::GenericOp op,
+                                PatternRewriter &rewriter) const override {
+    rewriter.startOpModification(op);
+    bool modifiedOutput = false;
+    Location loc = op.getLoc();
+    for (OpOperand &opOperand : op.getDpsInitsMutable()) {
+      if (!op.payloadUsesValueFromOperand(&opOperand)) {
+        Value operandVal = opOperand.get();
+        auto operandType = dyn_cast<RankedTensorType>(operandVal.getType());
+        if (!operandType) continue;
+
+        // If outs is sparse, leave it to the sparsifier.
+        if (sparse_tensor::getSparseTensorEncoding(operandVal.getType()))
+          continue;
+
+        // If outs is already an `empty` operation, nothing to do.
+        auto definingOp = operandVal.getDefiningOp<tensor::EmptyOp>();
+        if (definingOp) continue;
+        modifiedOutput = true;
+        SmallVector<OpFoldResult> mixedSizes =
+            tensor::getMixedSizes(rewriter, loc, operandVal);
+        Value emptyTensor = rewriter.create<tensor::EmptyOp>(
+            loc, mixedSizes, operandType.getElementType());
+        op->setOperand(opOperand.getOperandNumber(), emptyTensor);
+      }
+    }
+    if (!modifiedOutput) {
+      rewriter.cancelOpModification(op);
+      return failure();
+    }
+    rewriter.finalizeOpModification(op);
+    return success();
+  }
+};
+
+void populateElementwiseOpsFusionPatterns(RewritePatternSet &patterns) {
```

Comment:
Rename this since we are not doing fusion. You dont need the populate method. Just add the pattern directly below.